### DISCUSSION TITLE
[Feature]  Services dropdown

### DISCUSSION THREAD

**Discussion by ngocjohn** [ngocjohn/vehicle-status-card#15] [Ideas]
### Short summary

Add services sub card

### Detailed description

> > @rkkhanna The card tearing problem is solved, the fix is ready in the new release. Let me know how you like the new card.. 😉
> 
> Hi @ngocjohn, first views and some observations for the new `vehicle-status-card` configured with my Model X.
> 
> ![image](https://private-user-images.githubusercontent.com/57277191/369599607-818b9fcd-9863-4a14-b627-81bf274d178f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5NDI5MzEsIm5iZiI6MTcyNjk0MjYzMSwicGF0aCI6Ii81NzI3NzE5MS8zNjk1OTk2MDctODE4YjlmY2QtOTg2My00YTE0LWI2MjctODFiZjI3NGQxNzhmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDE4MTcxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlMzM2ZTMyMTI2ZDJkODQ0NThjYjNmNmMxZGI2YzQzOWYzZmU0NWQ3NzRkMmUwYjQzZDI2NjRjZjQ1YTllMDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.K4SwLiQE6pftvgTAsTeXoZ5LOWEpeHzfzA4FpVXdEnY)
> 
> 1. Check out the single indicators bar at the top. You will notice last icon is `mdi:account-off-outline` and the state label says `None`. Tesla has a binary sensor `binary_sensor.anju_model_x_user_present` which provides the user-id in an attribute when sensor is `on`. My intent is to show this indicator only when this binary sensor is `on`, and then show specific driver in the vehicle. While I already have code in `icon template` and `state template` to show who is driving the vehicle, is it possible to enhance card to include `visibility template` for single indicators similar to what you have provided for group indicators? See the screenshot of what I would show only when binary sensor is on:
> 
> ![image](https://private-user-images.githubusercontent.com/57277191/369600352-6a46aa2d-821b-4b96-82f7-60c6619a0ba2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5NDI5MzEsIm5iZiI6MTcyNjk0MjYzMSwicGF0aCI6Ii81NzI3NzE5MS8zNjk2MDAzNTItNmE0NmFhMmQtODIxYi00Yjk2LTgyZjctNjBjNjYxOWEwYmEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDE4MTcxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1ZDkwZDEwNmQyM2IzNWY5YzkxNmNjNDJmMTc1YWNlMGU3ODk0N2Y2NjUzZGFkMDlhN2IyNDU3Y2MyMWJkMzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bEk_FaJfpk2LXOaKmDaMAJ64Hi6_LsUcDPQBcGpG4J0)
> 
> 2. I also noticed there is some issue with the `Images` portion. I was unable to add multiple images through visual editor. I had to switch to YAML code editor view to add more images and then revert back to visual editor.
> 3. I couldn't figure out how to configure the `Services` dropdown through visual editor. Not sure if I missed it or if the card doesn't have the functionality for now. I guess I could use one of the buttons with a custom card such as mushroom-template to create buttons for controls provided by Tesla integration, e.g. lock/unlock, climate control, open/close trunk, etc. But it will be great to have services at the top the way you have it in `vehicle-info-card`. Your thoughts?
> 4. Last item I attempted tonight was to create a custom tire pressure display, similar to what I have for the Mercedes in your `vehicle-info-card`. See the two images below - left one is the tire display for Mercedes while the one on the right is for Tesla in the new card. Tesla is weird that the integration loses TPMS values soon after vehicle is turned off. Hence it is showing `unknown` value. Would it be possible to include `visibility template` at button level? This way I can make tire button visible only when integration is reporting valid tire pressure values.
> 
> ![image](https://private-user-images.githubusercontent.com/57277191/369601841-72678bc3-8baa-43c3-9b3f-86505356f2cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5NDI5MzEsIm5iZiI6MTcyNjk0MjYzMSwicGF0aCI6Ii81NzI3NzE5MS8zNjk2MDE4NDEtNzI2NzhiYzMtOGJhYS00M2MzLTliM2YtODY1MDUzNTZmMmNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDE4MTcxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxYjNkZDNkMDZjMjM5MjM5ZWZhMDI4ZGU1NGVjZjEyMWFkYTUzZGJmN2I2NDFkOWFmYTEyMDBjODliYzY0MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3VIQ4cpoQNszYyf4V9i9fxOyl2iCiWtCcz3qW_fHOjs)
> 
> Lastly, thank you for your work on this card. It's phenomenal!

**Comment by ngocjohn** [comment]
> 1. My intent is to show this indicator only when this binary sensor is `on`, and then show specific driver in the vehicle. While I already have code in `icon template` and `state template` to show who is driving the vehicle, is it possible to enhance card to include `visibility template` for single indicators similar to what you have provided for group indicators? See the screenshot of what I would show only when binary sensor is on

Yes, i will add the visibility condition for single indicators as well 👍 

> 2\. I also noticed there is some issue with the `Images` portion. I was unable to add multiple images through visual editor. I had to switch to YAML code editor view to add more images and then revert back to visual editor.

I'll check it out, I haven't noticed such a problem yet

> 3\. I couldn't figure out how to configure the `Services` dropdown through visual editor. Not sure if I missed it or if the card doesn't have the functionality for now. I guess I could use one of the buttons with a custom card such as mushroom-template to create buttons for controls provided by Tesla integration, e.g. lock/unlock, climate control, open/close trunk, etc. But it will be great to have services at the top the way you have it in `vehicle-info-card`. Your thoughts?

I prepared Tap actions for buttons, I will try to apply for indicators with possibility to switch to certain sub tab 🤔 

> 4\. Last item I attempted tonight was to create a custom tire pressure display, similar to what I have for the Mercedes in your `vehicle-info-card`. See the two images below - left one is the tire display for Mercedes while the one on the right is for Tesla in the new card. Tesla is weird that the integration loses TPMS values soon after vehicle is turned off. Hence it is showing `unknown` value. Would it be possible to include `visibility template` at button level? This way I can make tire button visible only when integration is reporting valid tire pressure values.

I think you can use conditional card, for example if it is off, show other lovelace etc. 😉

**Comment by horfic** [comment]
2. I can also confirm this issue. I only had 1 image for a longer period and wanted to add now additional ones. UI only allowed to delete. Adding multiple was only possible by code editor

**Comment by ngocjohn** [comment]
I have tested uploading images, both by drag and drop and by selecting from file. All methods work.. 🤷🏻‍♂️ 
![images-upload](https://github.com/user-attachments/assets/5b42da6e-a7a4-47b1-80a5-598c25cfa5fd)

**Comment by horfic** [comment]
![image](https://github.com/user-attachments/assets/a48fb2e8-9491-4db3-be4c-8efeaff679af)
![image](https://github.com/user-attachments/assets/d7409b82-79cc-4513-a830-9983165f064a)
![image](https://github.com/user-attachments/assets/16304941-94fd-49aa-99c4-33efd8db24c1)
![image](https://github.com/user-attachments/assets/800361f5-15aa-4790-98c7-0a22cc066fb5)


i press the add image button, paste an url into the input and than it loads the image on right and replaces the current single image path. But it does not add an additional one

**Comment by ngocjohn** [comment]
Oh, I get it. I forgot about the input form. I just left it for upload... I'll add it again in the new release. 😅

@horfic done! and new card type Tire pressures info is ready for release! 🚀 https://github.com/ngocjohn/vehicle-status-card/issues/3#issuecomment-2365453748
 
![fix-image-url](https://github.com/user-attachments/assets/3d30b4a5-e747-4af8-bc3a-358f6e73fdde)

**Comment by horfic** [comment]
nice, yes i'm already waiting for a release 1.1.0 so i can test it with hacs install

thank you very much.

if you need more testing or want to get more infos bmw connected api ha integration possibilities, i'm happy to help from the bmw side to a mercedes driver^^
